bin/node/runtime/src/lib.rs

```diff
@@ -1273,6 +1287,29 @@ impl_runtime_apis! {
 		}
 	}
 
+	impl pallet_mmr::primitives::MmrApi<
+		Block,
+		mmr::Leaf,
+		mmr::Hash,
+	> for Runtime {
+		fn generate_proof(leaf_index: u64) -> Result<(mmr::Leaf, mmr::Proof<mmr::Hash>), mmr::Error> {
+			Mmr::generate_proof(leaf_index)
+		}
+
+		fn verify_proof(leaf: mmr::Leaf, proof: mmr::Proof<mmr::Hash>) -> Result<(), mmr::Error> {
+			Mmr::verify_leaf(leaf, proof)
+		}
+
+		fn verify_proof_stateless(
+			root: mmr::Hash,
+			leaf: Vec<u8>,
```

Comment:
by curiosity: am I correct saying that user could also give the hash instead of the encoded leaf here ?

Reply:
So the leaf has to be encoded in it's `compact` form, so depending on the types used in the pallet it either has to be the hash (if it's wrapped in `Compact`) or the full leaf. I don't think passing a hash here would work in case your MMR leaf is configured without `Compact` (i.e. the "compact" encoding and regular encoding is the same).